src/System.Private.ServiceModel/tests/Common/Infrastructure/xunit/WcfTestCase.cs

```diff
@@ -40,11 +43,41 @@ internal WcfTestCase(IXunitTestCase testCase, string skippedReason)
 
         public void Deserialize(IXunitSerializationInfo info) { _testCase.Deserialize(info); }
 
-        public Task<RunSummary> RunAsync(
+        public async Task<RunSummary> RunAsync(
             IMessageSink diagnosticMessageSink, IMessageBus messageBus, object[] constructorArguments,
             ExceptionAggregator aggregator, CancellationTokenSource cancellationTokenSource)
         {
-            return new XunitTestCaseRunner(this, DisplayName, _skippedReason, constructorArguments, TestMethodArguments, messageBus, aggregator, cancellationTokenSource).RunAsync();
+            ConcurrentQueue<EventWrittenEventArgs> events = new ConcurrentQueue<EventWrittenEventArgs>();
+	          s_testListener._eventWritten = events.Enqueue;
+
+            var runsummary = await (new XunitTestCaseRunner(this, DisplayName, _testCase.SkipReason, constructorArguments, TestMethodArguments, messageBus, aggregator, cancellationTokenSource)).RunAsync();
```

Comment:
@hongdai -- I discovered we had a bug in WcfTestCase that I have fixed in PR #1433.  We cannot use XunitTestCaseRunner for tests marked with WcfTheory or it will fail.  It's not a difficult change, so we could potentially merge mine first and apply the fix to this PR.  But if you want to go first, I will update my PR with your changes here.  In either case, we can't move to WcfTheory until my fixes are in one of these PR's.